Gems/Atom/Feature/Common/Code/Source/CoreLights/PolygonLightFeatureProcessor.cpp

```diff
@@ -228,26 +228,27 @@ namespace AZ::Render
     {
         AZ_Warning("PolygonLightFeatureProcessor", vertexCount <= MaxPolygonPoints, "Too many polygon points on polygon light. Only using the first %lu vertices.", MaxPolygonPoints);
         AZ_Warning("PolygonLightFeatureProcessor", vertexCount >= 2, "Polygon light must have at least three points - ignoring points.");
-        
+
+        PolygonLightData& data = m_lightData.GetData<0>(handle.GetIndex());
         if (vertexCount < 3)
         {
-            return; // not enough points
+            data.SetEndIndex(data.GetStartIndex());
         }
-
-        PolygonPoints& pointArray = m_lightData.GetData<1>(handle.GetIndex());
-        uint32_t clippedCount = AZ::GetMin<uint32_t>(vertexCount, MaxPolygonPoints);
-        for (uint32_t i = 0; i < clippedCount; ++i)
+        else
         {
-            pointArray.at(i).x = vertices[i].GetX();
-            pointArray.at(i).y = vertices[i].GetY();
-            pointArray.at(i).z = vertices[i].GetZ();
+            PolygonPoints& pointArray = m_lightData.GetData<1>(handle.GetIndex());
+            uint32_t clippedCount = AZ::GetMin<uint32_t>(vertexCount, MaxPolygonPoints);
+            for (uint32_t i = 0; i < clippedCount; ++i)
+            {
+                pointArray.at(i).x = vertices[i].GetX();
+                pointArray.at(i).y = vertices[i].GetY();
+                pointArray.at(i).z = vertices[i].GetZ();
```

Comment:
Is there a way to avoid calling `at(i)` 3 times? Each one tests `i` to see if it's within the pointArray.

Reply:
When optimized the compiler doesn't do those checks because it has enough information to know that clippedCount can't go out of bounds. I can add an extra line that creates a ref to pointArray.a(i) though.

Reply:
But I can eliminate some code with StoreToFloat4 😄